server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceConfigNewSecretOwnershipStrategyInferTest.java

```diff
@@ -0,0 +1,94 @@
+import java.io.IOException;
+import java.util.Base64;
+import java.util.UUID;
+import keywhiz.IntegrationTestRule;
+import keywhiz.KeywhizService;
+import keywhiz.TestClients;
+import keywhiz.api.automation.v2.CreateGroupRequestV2;
+import keywhiz.api.automation.v2.CreateSecretRequestV2;
+import keywhiz.api.automation.v2.SecretDetailResponseV2;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SecretResourceConfigNewSecretOwnershipStrategyInferTest {
+  private static final ObjectMapper mapper = KeywhizService.customizeObjectMapper(Jackson.newObjectMapper());
+  private static final Base64.Encoder encoder = Base64.getEncoder();
+  private static final Integer RESPONSE_CODE_CREATED = 201;
+  private static final String AUTOMATION_CLIENT_GROUP_NAME = "Security";
+
+  private SecretResourceTest secretResourceTest;
+
+  OkHttpClient mutualSslClient;
+
+  @ClassRule public static final RuleChain chain = IntegrationTestRule.rule("keywhiz-test-newSecretOwnershipStrategyInfer.yaml");
+
+  @Before public void setUp() {
+    mutualSslClient = TestClients.mutualSslClient();
+    secretResourceTest = new SecretResourceTest();
+    secretResourceTest.mutualSslClient = mutualSslClient;
+  }
+
+  @Test
+  public void testCreateSecretWithNullOwnerName() throws IOException {
+    String secretName = randomString();
+    String content = randomString();
+    CreateSecretRequestV2 request = CreateSecretRequestV2.builder()
+        .name(secretName)
+        .content(encode(content))
+        .build();
+
+    Response response = secretResourceTest.create(request);
+    assertThat(response.code()).isEqualTo(RESPONSE_CODE_CREATED);
+
+    SecretDetailResponseV2 details = secretResourceTest.lookup(secretName);
+    assertThat(details.name()).isEqualTo(secretName);
+    assertThat(details.owner()).isEqualTo(AUTOMATION_CLIENT_GROUP_NAME);
+  }
+
+  @Test
+  public void testCreateSecretWithOwnerName() throws IOException {
+    String secretName = randomString();
+    String content = randomString();
+
+    String ownerName = randomString();
+    assertThat(createGroup(ownerName).code()).isEqualTo(RESPONSE_CODE_CREATED);
+
+    CreateSecretRequestV2 request = CreateSecretRequestV2.builder()
+        .name(secretName)
+        .owner(ownerName)
+        .content(encode(content))
+        .build();
+
+    Response response = secretResourceTest.create(request);
+    assertThat(response.code()).isEqualTo(RESPONSE_CODE_CREATED);
+
+    SecretDetailResponseV2 details = secretResourceTest.lookup(secretName);
+    assertThat(details.name()).isEqualTo(secretName);
+    assertThat(details.owner()).isEqualTo(ownerName);
+  }
+
+  private String randomString() {
+    return UUID.randomUUID().toString();
+  }
```

Comment:
I must have written this exact same method over 1,000 times in my career.

Random data is nice because it ensures that you won't have any collisions if you're inserting things into, say, a persistent database (integration testing against staging, for example) no matter how many times you run the tests. However, since all GUIDs more or less look like gibberish, it can sometimes be handy to add a human-readable prefix to it to indicate that it's coming from an integration test (doesn't really apply here, but something I've run into before).

Reply:
makes sense. I wrote the method also by mirroring the original SecretResourceTest but seems like a little extra here since we are not really worried about collision and all we want is some name to write assertions